gnat2goto/driver/tree_walk.adb

```diff
@@ -2094,17 +2101,28 @@ package body Tree_Walk is
             Set_Lhs (Op_Geq, Adjusted_Value_Expr);
             Set_Type (I     => Lower_Bound,
                       Value => Bound_Type);
-            Set_Value (I     => Lower_Bound,
-                      Value => Load_Bound_In_Hex (Get_Lower_Bound (Bound_Type),
-                        Pos (Get_Width (Bound_Type))));
             Set_Rhs (Op_Geq, Adjusted_Lower_Bound);
             Set_Type (Op_Geq, Make_Bool_Type);
             Set_Lhs (Op_Leq, Adjusted_Value_Expr);
             Set_Type (I     => Upper_Bound,
                       Value => Bound_Type);
-            Set_Value (I     => Upper_Bound,
-                      Value => Load_Bound_In_Hex (Get_Upper_Bound (Bound_Type),
-                        Pos (Get_Width (Bound_Type))));
+
+            if Kind (Bound_Type) = I_Bounded_Signedbv_Type then
+               Set_Value (Lower_Bound,
+                          Load_Bound_In_Hex (Get_Lower_Bound (Bound_Type),
+                            Pos (Get_Width (Bound_Type))));
+               Set_Value (Upper_Bound,
+                          Load_Bound_In_Hex (Get_Upper_Bound (Bound_Type),
+                            Pos (Get_Width (Bound_Type))));
+            else
+               Set_Value (Lower_Bound,
+                          Load_Real_Bound_In_Hex (Get_Lower_Bound (Bound_Type),
+                                                  Get_Width (Bound_Type)));
```

Comment:
Why is it `Pos (Get_Width (Bound_Type))` in one case and just `Get_Width (Bound_Type)` in the other?

Reply:
Well the int->hex conversion wants bit-width as a `Pos` type. In the real->hex conversion I only use the bit-width in a direct case-analysis, thus I don't care about the type.